internal/timeout/timeout.go

```diff
@@ -83,3 +94,90 @@ func Parse(timeout string) Setting {
 
 	return DurationSetting(d)
 }
+
+// AllowedRange defines a range of allowed values for a timeout. The zero
+// value of this struct is an AllowedRange that allows any value.
+type AllowedRange struct {
+	min time.Duration
+	max time.Duration
+
+	// This allows us to have a useful zero
+	// value for the struct that allows all
+	// values.
+	maxSet bool
+}
+
+// NewAllowedRange creates an AllowedRange with the given min and
+// max allowed durations.
+func NewAllowedRange(min, max time.Duration) AllowedRange {
+	return AllowedRange{
+		min:    min,
+		max:    max,
+		maxSet: true,
```

Comment:
If `max` is not configured, the essentially it becomes `math.MaxInt64`? Is there an Envoy default if not set? 

Just curious since it sort of contradicts the comment above (https://github.com/projectcontour/contour/pull/2741/files#diff-555ffae2a78c3e333d329e38f9e252aeR258-R259).